# 라이브러리

In [1]:
import pandas as pd

from copy import deepcopy

import warnings
warnings.filterwarnings('ignore')

# 거주지역별 주택 소유 및 무주택 가구 수

## 데이터 불러오기

In [4]:
house = pd.read_csv('/content/drive/MyDrive/멀티캠퍼스/[D9&10] 데이터시각화 프로젝트_4조/Data/인구 데이터/거주지역별_주택소유_및_무주택_가구수_20152020(https:  kosis.kr statHtml statHtml.do?orgId=101&tblId=DT_1OH0402&conn_path=I2).csv', encoding='cp949')
house

거주지역별(1)        2015    2015.1  ...        2020    2020.1   2020.2
0   거주지역별(1)  총 가구(일반가구)   주택소유 가구  ...  총 가구(일반가구)   주택소유 가구   무주택 가구
1         전국    19111030  10698686  ...    20926710  11730171  9196539
2      서울특별시     3784490   1875295  ...     3982290   1928074  2054216
3      부산광역시     1335900    786367  ...     1405037    809380   595657
4      대구광역시      928528    533011  ...      985816    569624   416192
5      인천광역시     1045417    611847  ...     1147200    659295   487905
6      광주광역시      567157    322530  ...      599217    343671   255546
7      대전광역시      582504    320654  ...      631208    330682   300526
8      울산광역시      423412    264500  ...      444087    286134   157953
9    세종특별자치시       75219     40715  ...      139106     73011    66095
10       경기도     4384742   2433305  ...     5098431   2842747  2255684
11       강원도      606117    334031  ...      661039    378696   282343
12      충청북도      601856    348127  ...      678922    394482   284440
13      충청남도      796185    446466  ...      892222    517605   374617
14      전라북도      717311    421666  ...      755575    448731   306844
15      전라남도      720612    431275  ...      761518    464463   297055
16      경상북도     1062724    635608  ...     1131819    688989   442830
17      경상남도     1258487    769925  ...     1350155    850274   499881
18   제주특별자치도      220369    123364  ...      263068    144313   118755

[19 rows x 19 columns]

## 컬럼명 변경 및 불필요한 데이터 삭제

컬럼명 변경  
`2015`    → `2015(총가구)`  
`2015.1`  → `2015(주택소유가구)`   
`2015.2`  → `2015(무주택가구)`     
로 바꿔줌

In [ ]:
house.columns

In [5]:
# 먼저 거주지역별(1)  →  거주지역 으로 바꿔줌
house.rename(columns={'거주지역별(1)':'행정구역'}, inplace=True)

In [ ]:
house.columns[3]

In [6]:
n = 0  # 컬럼 위치를 나타내기 위한 변수
for Y in range(2015,2021):
  for _ in range(1,4) :  # 각 연도마다 3번 씩 반복  
    n+=1  # 다음 컬럼을 보기 위해
    if house.columns[n] == f'{Y}':
      house.rename(columns={f'{Y}':f'{Y}(총가구)'}, inplace=True)        # 원래 컬럼명이  '연도' 인 경우  →  연도(총가구)  로 변경

    if house.columns[n] == f'{Y}.1':
      house.rename(columns={f'{Y}.1':f'{Y}(주택소유)'}, inplace=True)  # 원래 컬럼명이  '연도.1' 인 경우  →  연도(주택소유가구)  로 변경

    if house.columns[n] == f'{Y}.2':
      house.rename(columns={f'{Y}.2':f'{Y}(무주택)'}, inplace=True)  # 원래 컬럼명이  '연도.2' 인 경우  →  연도(무주택가구)  로 변경
house.head(2)

행정구역   2015(총가구) 2015(주택소유)  ...   2020(총가구) 2020(주택소유) 2020(무주택)
0  거주지역별(1)  총 가구(일반가구)    주택소유 가구  ...  총 가구(일반가구)    주택소유 가구    무주택 가구
1        전국    19111030   10698686  ...    20926710   11730171   9196539

[2 rows x 19 columns]

이제 0행은 필요 없으니 삭제

In [7]:
house.drop(index=[0],inplace=True)
house.reset_index(drop=True, inplace=True)

In [8]:
house

행정구역 2015(총가구) 2015(주택소유)  ... 2020(총가구) 2020(주택소유) 2020(무주택)
0        전국  19111030   10698686  ...  20926710   11730171   9196539
1     서울특별시   3784490    1875295  ...   3982290    1928074   2054216
2     부산광역시   1335900     786367  ...   1405037     809380    595657
3     대구광역시    928528     533011  ...    985816     569624    416192
4     인천광역시   1045417     611847  ...   1147200     659295    487905
5     광주광역시    567157     322530  ...    599217     343671    255546
6     대전광역시    582504     320654  ...    631208     330682    300526
7     울산광역시    423412     264500  ...    444087     286134    157953
8   세종특별자치시     75219      40715  ...    139106      73011     66095
9       경기도   4384742    2433305  ...   5098431    2842747   2255684
10      강원도    606117     334031  ...    661039     378696    282343
11     충청북도    601856     348127  ...    678922     394482    284440
12     충청남도    796185     446466  ...    892222     517605    374617
13     전라북도    717311     421666  ...    755575     448731    306844
14     전라남도    720612     431275  ...    761518     464463    297055
15     경상북도   1062724     635608  ...   1131819     688989    442830
16     경상남도   1258487     769925  ...   1350155     850274    499881
17  제주특별자치도    220369     123364  ...    263068     144313    118755

[18 rows x 19 columns]

## 수도권, 지방 추가하기

In [9]:
int(sum( house.iloc[[1, 4, 9], 1].astype(int) )/3)
#round ( sum( house.iloc[[1, 4, 9], i].astype(int) )/3 ,2 )

3071549

In [10]:
# 추가할 행 생성
house.loc[18] = house.loc[0]
house.loc[19] = house.loc[0]   
house.loc[18, '행정구역'] = '수도권'  # 1,4,9
house.loc[19, '행정구역'] = '지방'    # 2,3,5,6,7,8,10,11,12,13,14,15,16,17

# 수도권, 지방 컬럼값 재정의
for i in range(1, len(house.columns)):
  house.iloc[18, i] =int(sum( house.iloc[[1, 4, 9], i].astype(int) )/3)
  #round ( house.iloc[[1, 4, 9], i].mean(), 2 )
  house.iloc[19, i] = int(sum( house.iloc[[2,3,5,6,7,8,10,11,12,13,14,15,16,17], i].astype(int) )/15)

# 인데스 재정의
house = house.reindex(index=[ 0, 18, 19, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17])
house.reset_index(drop=True, inplace = True) 

# 완성본 csv파일로 저장
pd.DataFrame(house).to_csv('/content/drive/MyDrive/멀티캠퍼스/[D9&10] 데이터시각화 프로젝트_4조/이경희/data/거주지역별 주택소유 및 무주택 가구 수/거주지역별 주택소유 및 무주택 가구수(전체 데이터).csv', index=False)

나중에 쓸 일이 있을까 싶어 각 카테고리 별로도 나눴습니다

## 각 카테고리 별로 나누기

### base생성


In [ ]:
base = pd.DataFrame(house['행정구역'])

### join
- 각 카테고리별로 자르고 결합하기
- ex) `df1.join(df2)`

#### 총 가구 수 

In [ ]:
# 총 가구 수만 뽑아오기
house_total = house.iloc[:,1::3]

# base와 join
house_total = base.join(house_total)  # 완성

# 컬럼명 년도만 나타내기
for col in house_total.columns[1:]:
  house_total.rename(columns={col:col[:4]}, inplace= True)

house_total

행정구역      2015      2016      2017      2018      2019      2020
0        전국  19111030  19367696  19673875  19979188  20343188  20926710
1       수도권   3071549   3110652   3165498   3228670   3308208   3409307
2        지방   1933827   1960229   1990083   2018157   2050783   2108366
3     서울특별시   3784490   3784705   3813260   3839766   3896389   3982290
4     부산광역시   1335900   1344170   1354401   1363608   1377030   1405037
5     대구광역시    928528    935753    948030    957516    968620    985816
6     인천광역시   1045417   1062828   1080285   1094749   1120576   1147200
7     광주광역시    567157    569372    575732    578559    587159    599217
8     대전광역시    582504    590698    597736    602175    609043    631208
9     울산광역시    423412    426027    428720    431391    437094    444087
10  세종특별자치시     75219     90413    104325    119029    129664    139106
11      경기도   4384742   4484424   4602950   4751497   4907660   5098431
12      강원도    606117    616346    620729    628484    633942    661039
13     충청북도    601856    617914    629073    640978    654713    678922
14     충청남도    796185    813730    834986    851124    864102    892222
15     전라북도    717311    724678    728871    732980    738307    755575
16     전라남도    720612    727197    733757    737406    741026    761518
17     경상북도   1062724   1076581   1087807   1094534   1102934   1131819
18     경상남도   1258487   1273523   1292998   1306394   1321213   1350155
19  제주특별자치도    220369    229337    240215    248998    253716    263068

#### 주택 소유 가구 

In [ ]:
# 주택 소유 가구 수만 뽑아오기
householder = house.iloc[:,2::3]
# base와 join
householder = base.join(householder)

# 컬럼명 년도만 나타내기
for col in householder.columns[1:]:
  householder.rename(columns={col:col[:4]}, inplace= True)

householder

행정구역      2015      2016      2017      2018      2019      2020
0        전국  10698686  10743492  11000007  11233906  11456266  11730171
1       수도권   1640149   1649282   1682688   1723084   1763502   1810038
2        지방   1098461   1102609   1130129   1153237   1174801   1202015
3     서울특별시   1875295   1864059   1875189   1884423   1894875   1928074
4     부산광역시    786367    783542    791489    796271    799696    809380
5     대구광역시    533011    537709    550374    557967    563282    569624
6     인천광역시    611847    617040    630228    642745    650209    659295
7     광주광역시    322530    324437    328263    333853    338980    343671
8     대전광역시    320654    320525    320407    325228    326384    330682
9     울산광역시    264500    266518    271099    274789    279906    286134
10  세종특별자치시     40715     47215     55925     64544     69400     73011
11      경기도   2433305   2466749   2542649   2642084   2745423   2842747
12      강원도    334031    338824    345955    353035    363552    378696
13     충청북도    348127    351451    362726    371224    382876    394482
14     충청남도    446466    456199    477532    492564    504574    517605
15     전라북도    421666    415291    427522    434101    439491    448731
16     전라남도    431275    421262    435332    441640    449851    464463
17     경상북도    635608    632790    652416    665822    675292    688989
18     경상남도    769925    773636    800655    816442    831804    850274
19  제주특별자치도    123364    126245    132246    137174    140671    144313

#### 무주택 가구 수

In [ ]:
# 무주택 가구 수만 뽑아오기
homeless = house.iloc[:,3::3]

# base와 join
homeless = base.join(homeless)

# 컬럼명 년도만 나타내기
for col in homeless.columns[1:]:
  homeless.rename(columns={col:col[:4]}, inplace= True)

## CSV 형태로 변환

In [ ]:
# 전체 데이터
pd.DataFrame(house).to_csv('/content/drive/MyDrive/멀티캠퍼스/[D9&10] 데이터시각화 프로젝트_4조/이경희/data/거주지역별 주택소유 및 무주택 가구 수/거주지역별 주택소유 및 무주택 가구수(전체 데이터).csv', index=False)
# 총 가구수 
pd.DataFrame(house_total).to_csv('/content/drive/MyDrive/멀티캠퍼스/[D9&10] 데이터시각화 프로젝트_4조/이경희/data/거주지역별 주택소유 및 무주택 가구 수/거주지역별 주택소유 및 무주택 가구수(총가구수).csv', index=False)
# 주택 소유 가구 수
pd.DataFrame(householder).to_csv('/content/drive/MyDrive/멀티캠퍼스/[D9&10] 데이터시각화 프로젝트_4조/이경희/data/거주지역별 주택소유 및 무주택 가구 수/거주지역별 주택소유 및 무주택 가구수(주택소유).csv', index=False)
# 무주택 가구 수
pd.DataFrame(homeless).to_csv('/content/drive/MyDrive/멀티캠퍼스/[D9&10] 데이터시각화 프로젝트_4조/이경희/data/거주지역별 주택소유 및 무주택 가구 수/거주지역별 주택소유 및 무주택 가구수(무주택).csv',index=False)

# 성 연령대별 무주택 가구 수 

## 데이터 불러오기

In [ ]:
gender_age_homeless = pd.read_csv('/content/drive/MyDrive/멀티캠퍼스/[D9&10] 데이터시각화 프로젝트_4조/Data/인구 데이터/거주지역_가구주의_성_연령대별_무주택_가구수_20152020(httpskosis.krstatHtmlstatHtml.doorgId=101&tblId=DT_1OH0418&conn_path=I3).csv', encoding='cp949')
gender_age_homeless.head(3)

거주지역별(1)  성별(1)     2015   2015.1  ...   2020.4   2020.5  2020.6  2020.7
0  거주지역별(1)  성별(1)       총계    30세미만  ...   50~59세   60~69세  70~79세   80세이상
1        전국     총계  8412344  1161399  ...  1752476  1211581  641716  402397
2        전국     남자  5263961   632391  ...  1137161   746838  317331  127977

[3 rows x 50 columns]

## 컬럼명 변경 및 불필요한 데이터 삭제

- 컬럼명 변경  
`연도`    →  `총계`  
`연도.1`  →  `30세미만`  
`연도.2`  →  `30~39세`  
`연도.3`  →  `40~49세`  
`연도.4`  →  `50~59세`  
`연도.5`  →  `60~69세`  
`연도.6`  →  `70~79세`  
`연도.7`  →  `80세이상`  

In [ ]:
n = 1  # 컬럼 위치를 나타내기 위한 변수 (2열부터 시작하려고 1으로 초기화 함)
for Y in range(2015,2021):    # 연도
  for _ in range(8) :  #  각 연도마다 8개의 카테고리  
    n+=1  # 다음 컬럼을 보기 위해 
    if gender_age_homeless.columns[n] == f'{Y}':    gender_age_homeless.rename(columns={f'{Y}':f'{Y}(총계)'}, inplace=True)       
    if gender_age_homeless.columns[n] == f'{Y}.1':  gender_age_homeless.rename(columns={f'{Y}.1':f'{Y}(30세미만)'}, inplace=True)  
    if gender_age_homeless.columns[n] == f'{Y}.2':  gender_age_homeless.rename(columns={f'{Y}.2':f'{Y}(30~39세)'}, inplace=True)  
    if gender_age_homeless.columns[n] == f'{Y}.3':  gender_age_homeless.rename(columns={f'{Y}.3':f'{Y}(40~49세)'}, inplace=True) 
    if gender_age_homeless.columns[n] == f'{Y}.4':  gender_age_homeless.rename(columns={f'{Y}.4':f'{Y}(50~59세)'}, inplace=True) 
    if gender_age_homeless.columns[n] == f'{Y}.5':  gender_age_homeless.rename(columns={f'{Y}.5':f'{Y}(60~69세)'}, inplace=True) 
    if gender_age_homeless.columns[n] == f'{Y}.6':  gender_age_homeless.rename(columns={f'{Y}.6':f'{Y}(70~79세)'}, inplace=True) 
    if gender_age_homeless.columns[n] == f'{Y}.7':  gender_age_homeless.rename(columns={f'{Y}.7':f'{Y}(80세이상)'}, inplace=True) 
gender_age_homeless.head(1)

거주지역별(1)  성별(1) 2015(총계)  ... 2020(60~69세) 2020(70~79세) 2020(80세이상)
0  거주지역별(1)  성별(1)       총계  ...       60~69세       70~79세       80세이상

[1 rows x 50 columns]

- 컬럼명 변경   
`거주지역별(1)` → `거주지역`  
`성별(1)`  →  `성별`

In [ ]:
gender_age_homeless.rename(columns={'거주지역별(1)':'행정구역','성별(1)':'성별'}, inplace=True)

- 0행 삭제 & 인데스 재정의

In [ ]:
gender_age_homeless.drop(index=[0],inplace=True)
gender_age_homeless.reset_index(drop=True, inplace=True)

## 수도권, 지방 추가

In [ ]:
# 수도권    - 3,4,5, 12,13,14, 27,28,29
gender_age_homeless.loc[54] = gender_age_homeless.loc[0]   # 총계
gender_age_homeless.loc[55] = gender_age_homeless.loc[1]   # 남자
gender_age_homeless.loc[56] = gender_age_homeless.loc[2]   # 여자
gender_age_homeless.loc[54, '행정구역'] = '수도권'
gender_age_homeless.loc[55, '행정구역'] = '수도권'
gender_age_homeless.loc[56, '행정구역'] = '수도권'

# 지방 -  0,1,2,3,4,5,12,13,14,27,28,29 를 제외한 모든 값
gender_age_homeless.loc[57] = gender_age_homeless.loc[0]   # 총계
gender_age_homeless.loc[58] = gender_age_homeless.loc[1]   # 남자
gender_age_homeless.loc[59] = gender_age_homeless.loc[2]   # 여자
gender_age_homeless.loc[57, '행정구역'] = '지방'
gender_age_homeless.loc[58, '행정구역'] = '지방'
gender_age_homeless.loc[59, '행정구역'] = '지방'


# 각 해당 카테고리에 해당하는 인덱스 설정
지방_총계 = [6,9]
지방_총계 += [i for i in range(15,54,3)]
지방_남자 = [7,10]
지방_남자 += [i for i in range(16,54,3)]
지방_여자 = [8,11]
지방_여자 += [i for i in range(17,54,3)]

# 수도권, 지방 컬럼값 재정의
for i in range(2, len(gender_age_homeless.columns)):
  # 수도권
  gender_age_homeless.iloc[54, i] =int(sum( gender_age_homeless.iloc[[3, 12, 27], i].astype(int) )/3)  # 총계
  gender_age_homeless.iloc[55, i] =int(sum( gender_age_homeless.iloc[[4, 13, 28], i].astype(int) )/3)  # 남자
  gender_age_homeless.iloc[56, i] =int(sum( gender_age_homeless.iloc[[5, 14, 29], i].astype(int) )/3)  # 여자
  # 지방
  gender_age_homeless.iloc[57, i] =int(sum( gender_age_homeless.iloc[[1, 4, 9], i].astype(int) )/3) # 총계
  gender_age_homeless.iloc[58, i] =int(sum( gender_age_homeless.iloc[[1, 4, 9], i].astype(int) )/3) # 남자
  gender_age_homeless.iloc[59, i] =int(sum( gender_age_homeless.iloc[[1, 4, 9], i].astype(int) )/3) # 여자

# 인데스 재정의
idx = [0,1,2,54,55,56,57,58,59]
add = [i for i in range(3,54)]

gender_age_homeless = gender_age_homeless.reindex(index= idx + add )
gender_age_homeless.reset_index(drop=True, inplace = True) 

### 각 카테고리 별로 나누기

각 지역별 `총계`,`남자`,`여자`도 나중에 필요할지도 몰라서 따로 나눠서 저장했습니다

In [ ]:
len(gender_age_homeless)

54

#### 총계

In [ ]:
# 총계
gender_age_total = gender_age_homeless.iloc[::3,:]  # 총계만 봅아옴
gender_age_total.reset_index(drop=True,inplace=True)   # 잘랐으니까 다시 재정의 

gender_age_total.drop(columns='성별',inplace=True) # 성별 컬럼 삭제

#### 남자

In [ ]:
# 남자
gender_age_man = gender_age_homeless.iloc[1::3,:]  # 남자만 뽑아옴
gender_age_man.reset_index(drop=True,inplace=True)   # 잘랐으니까 다시 재정의 

gender_age_man.drop(columns='성별',inplace=True) # 성별 컬럼 삭제

#### 여자

In [ ]:
# 여자
gender_age_woman = gender_age_homeless.iloc[2::3,:]  # 여자 컬럼만 뽑아옴
gender_age_woman.reset_index(drop=True,inplace=True)   # 인덱스 재정의

gender_age_woman.drop(columns='성별',inplace=True) # 성별 컬럼 삭제

## CSV 형태로 변환

In [ ]:
# 전체 데이터
pd.DataFrame(gender_age_homeless).to_csv('/content/drive/MyDrive/멀티캠퍼스/[D9&10] 데이터시각화 프로젝트_4조/이경희/data/성 연령대 별 무주택 가구 수/성 연령대 별 무주택 가구 수(전체데이터).csv', index=False)
# 총계
pd.DataFrame(gender_age_total).to_csv('/content/drive/MyDrive/멀티캠퍼스/[D9&10] 데이터시각화 프로젝트_4조/이경희/data/성 연령대 별 무주택 가구 수/성 연령대 별 무주택 가구 수(총계).csv', index=False)
# 남자
pd.DataFrame(gender_age_man).to_csv('/content/drive/MyDrive/멀티캠퍼스/[D9&10] 데이터시각화 프로젝트_4조/이경희/data/성 연령대 별 무주택 가구 수/성 연령대 별 무주택 가구 수(남자).csv', index=False)
# 여자
pd.DataFrame(gender_age_woman).to_csv('/content/drive/MyDrive/멀티캠퍼스/[D9&10] 데이터시각화 프로젝트_4조/이경희/data/성 연령대 별 무주택 가구 수/성 연령대 별 무주택 가구 수(여자).csv', index=False)